dashboard/lib/services/curriculum_pdfs/resources.rb

```diff
@@ -182,9 +182,9 @@ def fetch_url_to_path(url, path)
             return path
           elsif url.start_with?("https://drive.google.com/")
             file_id = url_to_id(url)
-            file = service.getFile(file_id)
-            return nil unless file.export_links.include? "application/pdf"
-            service.export_file(file_id, 'application/pdf', download_dest: path)
+            file = service.get_file(file_id)
+            return nil unless file.mime_type == "application/pdf"
+            service.get_file(file_id, download_dest: path)
```

Comment:
I wonder if we should be using [`service.export_file`](https://googleapis.dev/ruby/google-api-client/latest/Google/Apis/DriveV3/DriveService.html#export_file-instance_method) instead. From the documentation it seems like `service.get_file` only "Gets a file's metadata or content by ID".

Reply:
It is not very clear since service.get_file also has a download_dest parameter...
```
Parameters:

file_id (String) — The ID of the file.
acknowledge_abuse (Boolean) (defaults to: nil) — Whether the user is acknowledging the risk of downloading known malware or other abusive files. This is only applicable when alt=media.
include_permissions_for_view (String) (defaults to: nil) — Specifies which additional view's permissions to include in the response. Only 'published' is supported.
supports_all_drives (Boolean) (defaults to: nil) — Whether the requesting application supports both My Drives and shared drives.
supports_team_drives (Boolean) (defaults to: nil) — Deprecated use supportsAllDrives instead.
fields (String) (defaults to: nil) — Selector specifying which fields to include in a partial response.
quota_user (String) (defaults to: nil) — An opaque string that represents a user for quota purposes. Must not exceed 40 characters.
user_ip (String) (defaults to: nil) — Deprecated. Please use quotaUser instead.
download_dest (IO, String) (defaults to: nil) — IO stream or filename to receive content download
options (Google::Apis::RequestOptions) (defaults to: nil) — Request-specific options
```

Reply:
It would make sense only if the files were already pdf files in the drive folder. This change seems to assume they are exactly pdfs being pulled (as opposed to exporting them as pdfs from other formats)?

In this case, it would just skip over any it finds that aren't actually pdfs.

Reply:
That does make this change make sense... even though it should, naturally, be able to export a pdf from a pdf, it also is more explicit and less complicated to just impose the restriction (as this change does) that only pdfs from google drive can be downloaded. That would match the comment that appears before this code block.

Reply:
Yeah, I also was thinking `export_file` was still the right move BUT I got an error when using that 

```
Error - #<Google::Apis::ClientError: Invalid request status_code: 403 header: #<HTTP::Message::Headers:0x000000029b5d23f0 @http_version="1.1", @body_size=0, @chunked=false, @request_method="GET", @request_uri=#<Addressable::URI:0x20c10 URI:https://www.googleapis.com/drive/v3/files/16EtSldvp1wgbS3M0tMOx2Y5rXkhfXHny/export?alt=media&mimeType=application%2Fpdf>, @request_query={"alt"=>"media", "mimeType"=>"application/pdf"}, @request_absolute_uri=nil, @status_code=403, @reason_phrase="Forbidden", @body_type=nil, @body_charset=nil, @body_date=nil, @body_encoding=#<Encoding:UTF-8>, @is_request=false, @header_item=[["X-GUploader-UploadID", "ABPtcPqvcwnl3xRMDLBSHDo_TV8GcMsDdwUpdIu3N70pAroD6Iz2TRQLYmiNDJN0r2sG7iLl62WIe3TpoQ"], ["Cache-Control", "private, max-age=0"], ["Content-Type", "application/json; charset=UTF-8"], ["Date", "Thu, 07 Mar 2024 16:58:53 GMT"], ["Expires", "Thu, 07 Mar 2024 16:58:53 GMT"], ["Server", "ESF"], ["Vary", "Origin, X-Origin"], ["X-Content-Type-Options", "nosniff"], ["X-Frame-Options", "SAMEORIGIN"], ["X-XSS-Protection", "0"], ["Content-Length", "264"], ["Alt-Svc", "h3=\":443\"; ma=2592000,h3-29=\":443\"; ma=2592000"]], @dumped=false>>
```

When I read the `export_file` documentation (below), I was kind of reading the following to mean a true "Google Doc" - not a "Google File".  It is interesting because no where else in the documentation does it reference "Google Doc".

> Exports a **Google Doc** to the requested MIME type and returns the exported content.

Since this file is already a PDF (based on the `mime_type`, the export functionality isn't as needed as a normal "download" - based on the documentation, I am not exactly sure if `get_file` is the same as "hey, download it"...  BUT using `get_file` did work on two different courses with this type of weird "It is a PDF in a Google Drive" type of organization.

Reply:
That makes sense, the `download_dest` in `get_file` receives the `File` content.

Reply:
Actually diving deeper into the source code [here](https://github.com/googleapis/google-api-ruby-client/blob/22714861072809b4f8fe1d4d6c457c02e3dc28a1/generated/google-apis-drive_v3/lib/google/apis/drive_v3/service.rb#L1131), I think I feel even better about using the `get_file` here.  Happy to do a live code review too to talk it out :)